packages/utilities/src/customizable.tsx

```diff
@@ -43,6 +43,11 @@ export function customizable(
               // tslint:disable-next-line:no-any
               const componentProps = this.props as any;
 
+              // If defaultProps.styles is a function, evaluate it before calling concatStyleSets
```

Comment:
This will support styles from context, but not from button, so the `IButtonProps.styles` type would not be modified. It's a stopgap solution for sure, but I think I like it a little better than touching BaseButton or any of that.

However... there is an issue here. This only would work with customizable things. What about things built with `styled`?

```js
const Customizations = {
  scopedSettings: {
    DefaultButton: { styles: buttonProps => ({ ... }) },
    Toggle: { styles: (toggleProps) => { ... } }
  }
}
```

Reply:
I think that would work, because IToggleProps.styles is an IStyleFunctionOrObject...

One other subtle asymmetry here is the typing issue. In the above example, `buttonProps` is `IButtonProps`, while `toggleProps` is not `IToggleProps` but rather `IToggleStyleProps`.

I *think* overall this PR fixes the issue but we need to, as soon as possible, nuke customizable and styled and replace with a singular pattern that just makes sense and is completely predictable and symmetrical in the above customization scenario.